tests/Benchmark.test.ts

```diff
@@ -51,111 +53,824 @@ const buildLSP6TestContext = async (
+              [
+                context.accounts[1].address,
+                context.accounts[2].address,
+                context.accounts[3].address,
+              ],
+              [
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+              ],
+              ["0x", "0x", "0x"]
+            );
+
+          const receipt = await tx.wait();
+
+          executeUP.push([
+            "transfer LYX from UP to 3x EOA without data",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+
+        it("Sending LYX from UP to 3 UP without data", async () => {
+          const tx = await context.universalProfile
+            .connect(context.owner)
+            ["execute(uint256[],address[],uint256[],bytes[])"](
+              [
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+              ],
+              [
+                universalProfile1.address,
+                universalProfile2.address,
+                universalProfile3.address,
+              ],
+              [
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+              ],
+              ["0x", "0x", "0x"]
+            );
+
+          const receipt = await tx.wait();
+
+          executeUP.push([
+            "transfer LYX from UP to 3x UP without data",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+
+        it("Sending LYX from UP to EOA with random data 256 bytes", async () => {
+          const tx = await context.universalProfile
+            .connect(context.owner)
+            ["execute(uint256[],address[],uint256[],bytes[])"](
+              [
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+              ],
+              [
+                context.accounts[1].address,
+                context.accounts[2].address,
+                context.accounts[3].address,
+              ],
+              [
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+              ],
+              [
+                ethers.utils.hexlify(ethers.utils.randomBytes(256)),
+                ethers.utils.hexlify(ethers.utils.randomBytes(256)),
+                ethers.utils.hexlify(ethers.utils.randomBytes(256)),
+              ]
+            );
+
+          const receipt = await tx.wait();
+
+          executeUP.push([
+            "transfer LYX from UP to 3x EOA with 256 bytes data",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+
+        it("Sending LYX from UP to UP with random data 256 bytes prefixed with 4 zeros", async () => {
+          const random256BytesData =
```

Comment:
I suggest avoid using `.substring(2)` and making the most of the ethersjs library

```suggestion
          ethers.utils.hexConcat([
              "0x00000000",
              ethers.utils.hexlify(ethers.utils.randomBytes(252))
          ])
```